File: web/src/app/models/Article.ts
import mongoose from 'mongoose';

const ArticleSchema = new mongoose.Schema({
  articleId: {
    type: String,
    required: true,
    unique: true,
  },
  title: {
    type: String,
    required: true,
  },
  slug: {
    type: String,
    required: true,
    unique: true,
  },
  metaTitle: {
    type: String,
    required: true,
  },
  metaDescription: {
    type: String,
    required: true,
  },
  description: {
    type: String,
  },
  createdDate: {
    type: Date,
    required: true,
    default: Date.now,
  },
  updatedDate: {
    type: Date,
    default: Date.now,
  },
  weekNumber: {
    type: String,
    required: true,
  },
  category: {
    type: String,
    required: true,
  },
  excerpt: {
    type: String,
    required: true,
  },
  author: {
    type: String,
    required: true,
    default: 'Anas Aboreeda',
  },
  authorImage: {
    type: String,
    default: 'https://avatars.githubusercontent.com/u/22588845?v=4',
  },
  tags: {
    type: [String],
    default: [],
  },
  views: {
    type: Number,
    default: 0,
  },
  claps: {
    type: Number,
    default: 0,
  },
  readTime: {
    type: String,
    required: true,
  },
  bannerImage: {
    type: String,
    required: true,
  },
  highlighted: {
    type: String,
    enum: ['PRIMARY', 'SECONDARY', 'NONE'],
    default: 'NONE',
  },
  published: {
    type: Boolean,
    default: false,
  },
  content: {
    type: String,
    required: true,
  },
});

export interface IArticle extends Document {
  articleId: string;
  title: string;
  slug: string;
  metaTitle: string;
  metaDescription: string;
  description: string;
  createdDate: Date;
  updatedDate: Date;
  weekNumber: string;
  category: string;
  excerpt: string;
  author: string;
  authorImage: string;
  tags: string[];
  views?: number;
  claps?: number;
  readTime: string;
  bannerImage: string;
  highlighted?: 'PRIMARY' | 'SECONDARY' | 'NONE';
  published: boolean;
  content: string;
}

export default mongoose.models.Article || mongoose.model('Article', ArticleSchema);

File: api/api/src/main/java/com/anascode/api/model/Article.java
package com.anascode.api.model;

import lombok.Data;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;
import org.springframework.data.mongodb.core.mapping.Field;

import java.util.Date;
import java.util.List;

@Data
@Document(collection = "articles")
public class Article {

    @Id
    private String articleId;

    private String title;

    private String slug;

    private String metaTitle;

    private String metaDescription;

    private String description;

    @Field("created_date")
    private Date createdDate;

    @Field("updated_date")
    private Date updatedDate;

    @Field("week_number")
    private String weekNumber;

    private String category;

    private String excerpt;

    private String author;

    @Field("author_image")
    private String authorImage;

    private List<String> tags;

    private int views;

    private int claps;

    @Field("read_time")
    private String readTime;

    @Field("banner_image")
    private String bannerImage;

    private Highlighted highlighted;

    private boolean published;

    private String content;

    public enum Highlighted {
        PRIMARY, SECONDARY, NONE
    }

}
